src/NLog.Targets.Syslog/MessageCreation/SdElement.cs

```diff
@@ -15,22 +14,22 @@ internal class SdElement
         private static readonly byte[] RightBracketBytes = { 0x5D };
 
         private readonly SdId sdId;
-        private readonly List<SdParam> sdParams;
+        private readonly IList<SdParam> sdParams;
```

Comment:
The reason for using `List<SdParam>` is because then there is no allocation of boxed-enumerator. List has a non-allocating struct-enumerator (`IList` does not).

Reply:
I reverted the change, but I think it is worth pointing out that performance improvements are meant to have a noticeable impact on most users and use cases, otherwise we will end up writing native code without mission critical or low latency needs: in this specific case, even if we had 10000 SdParams (and we are way below this figure) the [gain would be](https://stackoverflow.com/a/3168435) 50 ns.